File: app/src/main/java/com/solution/it/newsoft/NewSoftApp.java
package com.solution.it.newsoft;

import com.solution.it.newsoft.dagger.DaggerAppComponent;

import javax.inject.Singleton;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

@Singleton
public class NewSoftApp extends DaggerApplication {

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }
}

File: app/src/main/java/com/solution/it/newsoft/NewSoftApp.kt
package com.solution.it.newsoft

import android.app.Application
import com.solution.it.newsoft.koin.*

import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

class NewSoftApp : Application() {

    override fun onCreate() {
        super.onCreate()
        setupKoin()
    }

    private fun setupKoin() {
        startKoin {
            androidLogger()
            androidContext(this@NewSoftApp)
            modules(listOf(
                    appModule,
                    listingModule
            ))
        }
    }
}
